File: retailStore_BE/src/main/java/net/yorksolutions/optumfsjavadukesofyork2/models/Product.java
package net.yorksolutions.optumfsjavadukesofyork2.models;

import jakarta.persistence.*;

import java.time.LocalDate;
import java.util.List;

@Entity
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    public Long id;

    public String name;
    @OneToOne
    public Category category;
    public LocalDate availability;
    public String description;
    public boolean discontinued;
    public String image;
    @OneToMany(cascade = CascadeType.ALL)
    public List<Price> scheduledMaps;
    @OneToMany(cascade = CascadeType.ALL)
    public List<Price> scheduledPrices;
    @OneToMany(cascade = CascadeType.ALL)
    public List<Price> scheduledSales;
    @OneToMany(cascade = CascadeType.ALL)
    public List<Shipment> shipments;

    @ManyToMany
    public List<CustomerOrder> orders;

    public Product(){}
}

File: optum-fs-java-dukes_of_york-1-main/src/app/data/product.ts
import { Categories } from "./categories";
import { Price } from "./price";
import { Shipment } from "./shipments";

export class Product {
    public id : number | null = null;
    public name: string
    public discontinued: boolean
    public availability: Date
    public description: string
    public image: string
    public category: Categories | null
    public scheduledMaps: Price[]
    public scheduledPrices: Price[]
    public scheduledSales: Price[]
    public shipments: Shipment[]

    constructor(name: string, discontinued: boolean, availability: Date, description: string, image: string, category: Categories | null, scheduledMAPS: Price[], scheduledPrices: Price[], scheduledSales: Price[], shipments: Shipment[]){
        this.name = name
        this.discontinued = discontinued
        this.availability = availability
        this.description = description
        this.image = image
        this.category = category
        this.scheduledMaps = scheduledMAPS
        this.scheduledPrices = scheduledPrices
        this.scheduledSales = scheduledSales
        this.shipments = shipments
    }
}